File: recursion/javascript/powerOf.js
// calculate the power of a number recursively

function powerOf(base, exponent) {
    if (exponent === 0) {
        return 1;
    } else if (exponent > 0) {
        return (1 / base) * powerOf(base, exponent + 1);
    } else {
        return base * powerOf(base, exponent - 1);
    }
}

console.log(powerOf(2, 10));

File: recursion/python/power_of.py
# calculate the power of a number recursively


def power_of(base, exponent):

    assert int(exponent) == exponent, 'The exponent must be an integer number'

    if exponent == 0:
        return 1
    elif exponent < 0:
        return 1 / base * power_of(base, exponent + 1)
    else:
        return base * power_of(base, exponent - 1)


if __name__ == '__main__':
    for i in range(-5, 5):
        print(f'2**{i} = {power_of(2, i)}')
